pennylane/_qubit_device.py

```diff
@@ -1189,6 +1204,78 @@ def mutual_info(self, wires0, wires1, log_base):
             state, indices0=wires0, indices1=wires1, c_dtype=self.C_DTYPE, base=log_base
         )
 
+    def classical_shadow(self, wires, n_snapshots, circuit, seed=None):
+        """
+        Returns the measured bits and recipes in the classical shadow protocol.
+
+        The protocol is described in detail in the `classical shadows paper <https://arxiv.org/abs/2002.08953>`_.
+        This measurement process returns the randomized Pauli measurements (the ``recipes``)
+        that are performed for each qubit and snapshot as an integer:
+
+        - 0 for Pauli X,
+        - 1 for Pauli Y, and
+        - 2 for PauliZ.
+
+        It also returns the measurement results (the ``bits``); 0 if the 1 eigenvalue
+        is sampled, and 1 if the -1 eigenvalue is sampled.
+
+        The device shots are used to specify the number of snapshots. If ``T`` is the number
+        of shots and ``n`` is the number of qubits, then both the measured bits and the
+        Pauli measurements have shape ``(T, n)``.
+
+        This implementation is device-agnostic and works by executing single-shot
+        tapes containing randomized Pauli observables. Devices should override this
+        if they can offer cleaner or faster implementations.
+
+        .. seealso:: :func:`~.classical_shadow`
+
+        Args:
+            wires (Sequence[int]): The wires to perform Pauli measurements on
+            n_snapshots (int): The number of snapshots
+            circuit (~.tapes.QuantumTape): The quantum tape that is being executed
+            seed (Union[int, None]): If provided, it is used to seed the random
+                number generation for generating the Pauli measurements.
+
+        Returns:
+            tensor_like[int]: A tensor with shape ``(2, T, n)``, where the first row represents
+            the measured bits and the second represents the recipes used.
+        """
+        if circuit is None:  # pragma: no cover
+            raise ValueError("Circuit must be provided when measuring classical shadows")
+
+        with set_shots(self, shots=1):
+            # slow implementation but works for all devices
+            n_qubits = len(self.wires)
+            device_wires = np.array(self.map_wires(wires))
```

Comment:
Shouldn't this be `obs_wires`? Or maybe you wanted to do:

```suggestion
        device_wires = np.array(self.map_wires(self.wires))
```

Reply:
Same as above. This should be correct.